### PR TITLE
Add smoke test and stress test for the write path

### LINKED ISSUE #370: Add k6 script for mocking/smoke testing the write path
(from the Slack discussion)

It might be interesting to add a [k6](https://github.com/loadimpact/k6) script for testing the write path of Tempo. This script could be used for:
- Generating fake trace data and shipping it to Tempo.
- Smoke testing new/existing Tempo deployments.
- Stress/Soak testing the write path.

I propose (for this first script) that we use the HTTP Zipkin receiver. Later-on we can add the other receivers :)

If this experiment goes well, might be interesting to add another script for the read path.

### PR DESCRIPTION
**Pull request by dgzlopes**
**What this PR does**:
This PR adds two [k6](https://github.com/loadimpact/k6) tests: 

`smoke_test.js`
*Use case:* Verifies that Tempo can handle minimal load, without problems.

`stress_test_write_path.js`
*Use case:* Stresses Tempo's write path with different stages.

Both scripts use the [Zipkin HTTP](https://zipkin.io/zipkin-api/) receiver.

I'll like to thank @calvernaz! Loved your Go scripts and took some inspiration from them :)
And @simskij for helping with my newbie JS/k6 doubts.

**Next steps (?)**:
These scripts are generic and basic but can be quite handy for development or while testing new Tempo deployments. 

Down the road probably it's interesting if we:
- Think if we want to create or integrate existing tests on CI.
- Improve thresholds.
- Add more "specific" tests. For example, @simskij proposed a long-running Soak test, where we push spans slowly (to mock msg queues and/or ultra-long traces).
- Build tests using other receivers.
- etc

btw, I was happy to reproduce some open issues by tweaking these two scripts.

Also, I want to add, that I'm working on some [k6 extensions](https://k6.io/blog/extending-k6-with-xk6): Prometheus, Redis, and Memcached. That means that while we're running the test, we can read Prometheus metrics and/or interact with Tempo's caches. 

I mention this b/c, with those extensions, we probably can create some **fancy** tests.

**Which issue(s) this PR fixes**:
Fixes #370 

**Checklist**
- [ ] Tests updated
- [X] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by simskij — APPROVED** [review]
From a k6 perspective, this looks nice. 👏

File: tools/k6/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+If you're running Tempo locally in monolith mode (for e.g while developing), the scripts are ready to use.
+
+If you have the k6 binary installed:
+```bash
+$ k6 run script.js
+```
+
+If you're using the Docker image:
+```bash
+$ docker run -i loadimpact/k6 run - <script.js
```
Comment (joe-elliott) [inline tools/k6/README.md:48]: This command does not appear to work.  I was able to get it to work with the following with a locally running tempo:
```
docker run -v $PWD:/src -i --network host loadimpact/k6 run /src/smoke_test.js
```

**Review by joe-elliott — CHANGES_REQUESTED** [review]
This is extremely cool:

```
    ✓ ingester status is 200
    ✓ write status is 202
    ✓ tempo-query status is 200
    ✓ querier status is 200
    ✓ read status is 200
    ✓ distributor status is 200

    checks.....................: 100.00% ✓ 88  ✗ 0  
    ✓ { type:read }............: 100.00% ✓ 30  ✗ 0  
    ✓ { type:steady }..........: 100.00% ✓ 28  ✗ 0  
    ✓ { type:write }...........: 100.00% ✓ 30  ✗ 0  
    data_received..............: 19 kB   542 B/s
    data_sent..................: 11 kB   301 B/s
    http_req_blocked...........: avg=17.19µs min=1.41µs   med=10.25µs max=454.78µs p(90)=15.51µs  p(95)=18.15µs 
    http_req_connecting........: avg=5.46µs  min=0s       med=0s      max=283.83µs p(90)=0s       p(95)=0s      
  ✓ http_req_duration..........: avg=1.86ms  min=166.92µs med=1.54ms  max=4.13ms   p(90)=3.34ms   p(95)=3.42ms  
    http_req_receiving.........: avg=92.04µs min=22.72µs  med=92.88µs max=275.78µs p(90)=131.01µs p(95)=148.85µs
    http_req_sending...........: avg=55.57µs min=7.31µs   med=51.23µs max=145.98µs p(90)=85.54µs  p(95)=106.6µs 
    http_req_tls_handshaking...: avg=0s      min=0s       med=0s      max=0s       p(90)=0s       p(95)=0s      
    http_req_waiting...........: avg=1.71ms  min=133.44µs med=1.44ms  max=3.76ms   p(90)=3.13ms   p(95)=3.24ms  
    http_reqs..................: 88      2.510376/s
    iteration_duration.........: avg=1.42s   min=1s       med=1s      max=5s       p(90)=2.6s     p(95)=5s      
    iterations.................: 67      1.911309/s
    vus........................: 1       min=1 max=3
    vus_max....................: 3       min=3 max=3
```

Can we move this to `./integration/k6` ?  I see this being part of our integration tests and it makes more sense to me there.

File: tools/k6/README.md (hunk 1 above)
Comment (dgzlopes) [reply tools/k6/README.md:48]: Oh, right - if we use the Docker container we need to access the host's network. I'll change it, nice catch :)

**Review by joe-elliott — APPROVED** [review]

**Pull request merged**